cdisc_rules_engine/check_operators/dataframe_operators.py

```diff
@@ -1643,6 +1654,41 @@ def check_date_overlaps(self, group, target, comparator):
 
         return is_sorted
 
+    def _process_grouped_result(
+        self,
+        grouped_result,
+        grouped_df,
+        within_columns,
+        sorted_df,
+        is_pandas_dataset,
+        check_func=None,
+    ):
+        if isinstance(grouped_result, pd.DataFrame):
+            grouped_result = grouped_result.iloc[:, 0]
+        if is_pandas_dataset and isinstance(grouped_result.index, pd.MultiIndex):
```

Comment:
whenever you have to check isintance(pandas), it is usually a sign that this should be handled by the [DatasetInterface](https://github.com/cdisc-org/cdisc-rules-engine/blob/f38215e4a067f5118c83e4795d6e94d6b18fd0b8/cdisc_rules_engine/models/dataset/dataset_interface.py)